applications/luci-app-throughput-measurements/root/usr/libexec/rpcd/luci.tputmeas

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env lua
+
+local ubus = require "ubus"
+local uci = require "luci.model.uci"
+local nixio = require "nixio"
+local fs = require "nixio.fs"
+local json = require "luci.jsonc"
+local sys = require "luci.sys"
+local posix = require "posix"
+
+
+local methods = {
+	iperf3Start = {
+		args = { mode = "mode", server_address = "server_address", interface = "interface",
+				interface_address = "interface_address", port = "port", duration = "duration",
+				bytes = "bytes", udp = "udp",  bitrate = "bitrate", reverse = "reverse",
+				connections = "connections", interval = "interval" },
+		call = function(args)
+			local iperf3_cmd = "iperf3 --forceflush "
+
+			if fs.stat("/var/run/iperf3.pid") then
+				return { result = "error: Iperf3 is running" }
+			else
+				if not args then
+					return { result = "error: Lack of arguments" }
+				end
+
+				if (not args.mode or (args.mode ~= "-c" and args.mode ~= "-s")) then
+					return { result = "error: mode has not been specified " }
+				end
+
+				iperf3_cmd = iperf3_cmd .. args.mode
+
+				if args.server_address and args.server_address ~= '' then
+					local ipparts = {args.server_address:match("^(%d+)%.(%d+)%.(%d+)%.(%d+)$")}
```

Comment:
You could use [`luci.ip.new()`](http://openwrt.github.io/luci/api/modules/luci.ip.html#new), which would also allow IPv6. If you indeed need to enforce IPv4, use [`luci.ip.IPv4()`](http://openwrt.github.io/luci/api/modules/luci.ip.html#IPv4) instead.

```lua
local addr = luci.ip.new(args.server_address)
if addr then
    iperf3_cmd = iperf3_cmd .. " " .. addr:host()
else
    return { error = "Invalid server address" }
end
```